sdk/executor/src/main/java/com/mesosphere/sdk/executor/ProcessTask.java

```diff
@@ -62,6 +62,19 @@ protected ProcessTask(
         this.taskInfo = taskInfo;
         this.processBuilder = processBuilder;
         this.exitOnTermination = exitOnTermination;
+        this.taskKillGracePeriod = getTaskKillGracePeriod(taskInfo);
+    }
+
+    private Duration getTaskKillGracePeriod(Protos.TaskInfo taskInfo) {
```

Comment:
Thank you much. I was here and below again following how I understood what is the consistent way of handling optional protobuf fields. Your clarification here and below to base such decisions upon whether the model field is `Optional<>` helps a bunch.